staging/src/k8s.io/client-go/discovery/cached/disk/compactable_disk_cache.go

```diff
@@ -0,0 +1,236 @@
+
+	"golang.org/x/sync/singleflight"
+
+	"github.com/peterbourgon/diskv"
+	"k8s.io/klog/v2"
+)
+
+const oneDay = 24 * time.Hour
+
+// CompactableDiskCache implements the httpcache.Cache interface. It delegates to
+// diskv.Diskv for Cache functions (Get, Set, Delete), and occasionally removes
+// old cached files. The specified "cacheSize" is NOT a hard limit; the disk
+// cache can breach this threshold. When cache compaction occurs, it will resize
+// the disk cache below this "cacheSize".
+//
+// Trigger: disk cache compactions are triggered when the modification time
+//   of an accessed cache file is older than one day. In this case, the
+//   accessed file's modification time is updated to the current time
+//   (file touched), and the cache compaction is run in parallel. Access
+//   time is not used directly because it is not well supported by all OSes.
+//
+// Concurrency:
+//   Within the same process: the goroutines of a single process are
+//   coordinated using "sync/singleflight" functionality.
+//
+//   Among different processes: if (for example) a separate kubectl command
+//   compacts the disk cache at the same time, separate processes can compete
+//   to delete the same files. This code must be tolerant to attempting to
+//   delete files already deleted.
+//
+type CompactableDiskCache struct {
+	cacheDir  string
+	cacheSize int64
+	diskv     *diskv.Diskv
+}
+
+// Get returns the []byte representation of a cached response and a bool
+// set to true if the value isn't empty. This function can trigger a
+// disk cache compaction which runs in a separate goroutine.
+func (tds *CompactableDiskCache) Get(key string) (responseBytes []byte, ok bool) {
+	filename := keyToFilename(key)
+	go possiblyCompactDiskCache(tds.cacheDir, filename, tds.cacheSize)
+	resp, err := tds.diskv.Read(filename)
+	if err != nil {
+		return []byte{}, false
+	}
+	return resp, true
+}
+
+// Set stores the []byte representation of a response against a key.
+func (tds *CompactableDiskCache) Set(key string, responseBytes []byte) {
+	key = keyToFilename(key)
+	tds.diskv.WriteStream(key, bytes.NewReader(responseBytes), true)
+}
+
+// Delete removes the value associated with the key.
+func (tds *CompactableDiskCache) Delete(key string) {
+	key = keyToFilename(key)
+	tds.diskv.Erase(key)
+}
+
+// NewCompactableDiskCache returns a newly created CompactableDiskCache,
+// which includes a newly created diskv.Diskv delegate.
+func NewCompactableDiskCache(cacheDir string, cacheSize int64) *CompactableDiskCache {
+	return &CompactableDiskCache{
+		cacheDir:  cacheDir,
+		cacheSize: cacheSize,
+		diskv: diskv.New(diskv.Options{
+			PathPerm: os.FileMode(0750),
+			FilePerm: os.FileMode(0660),
+			BasePath: cacheDir,
+			TempDir:  filepath.Join(cacheDir, ".diskv-temp"),
+		}),
+	}
+}
+
+// IMPORTANT: This function MUST be the same as
+// "github.com/gregjones/httpcache/diskcache.keyToFilename()" in
+// order to maintain compatibility with existing http disk caches.
```

Comment:
What would the consequence of this function changing be?

Reply:
It appears that Greg Jones is using the MD5 hash to map a URL to a filename. By changing the hash algorithm, all existing disk-cached files will no longer be found.